mlir/lib/Dialect/Vector/Transforms/LowerVectorTransfer.cpp

```diff
@@ -132,7 +131,7 @@ struct TransferReadPermutationLowering
     }
 
     // Transpose in_bounds attribute.
-    ArrayAttr newInBoundsAttr =
+    DenseBoolArrayAttr newInBoundsAttr =
         inverseTransposeInBoundsAttr(rewriter, op.getInBounds(), permutation);
```

Comment:
Also here, I think `inverseTransposeInBoundsAttr` can be removed, and instead we can now just use the existing `applyPermutation`:  

```
SmallVector<bool> newInBounds = applyPermutation(op.getInBounds(), permutation);
```

(and just delay constructing the `DenseBoolArrayAttr`)

Reply:
Updated.